src/server/active_tran_server.cpp

```diff
@@ -351,23 +369,34 @@ void active_tran_server::receive_data_page (cubpacking::unpacker &upk)
   std::string message;
   upk.unpack_string (message);
 
-  int error_code;
-  std::memcpy (&error_code, message.c_str (), sizeof (error_code));
+  if (message.size () == sizeof (int))
```

Comment:
A comment could help
```suggestion
  // The message is either an error code or the content of the data page
  if (message.size () == sizeof (int))
```